pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -39,24 +41,36 @@ var currentExecutionInfo executionStartInfo
 
 // startExecutionSpan records information from the start of the invocation.
 // It should be called at the start of the invocation.
-func startExecutionSpan(startTime time.Time, rawPayload string) {
+func startExecutionSpan(startTime time.Time, rawPayload string, inferredSpan inferredspan.InferredSpan) {
 	currentExecutionInfo.startTime = startTime
-	currentExecutionInfo.traceID = random.Uint64()
-	currentExecutionInfo.spanID = random.Uint64()
+	currentExecutionInfo.traceID = rand.Random.Uint64()
+	currentExecutionInfo.spanID = rand.Random.Uint64()
 	currentExecutionInfo.parentID = 0
 
 	payload := convertRawPayload(rawPayload)
 
+	if inferredspan.IsExpected() {
+		currentExecutionInfo.traceID = inferredSpan.Span.TraceID
+		currentExecutionInfo.parentID = inferredSpan.Span.SpanID
+	}
+
 	if payload.Headers != nil {
 		traceID, e1 := convertStrToUnit64(payload.Headers[TraceIDHeader])
```

Comment:
Nit: It's not a new thing but should probably fix the typo here: `Unit64` -> `Uint64`